Android/app/src/main/java/app/intra/sys/firebase/AnalyticsWrapper.java

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2019 Jigsaw Operations LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package app.intra.sys.firebase;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import app.intra.sys.firebase.BundleBuilder.Params;
+import com.google.firebase.analytics.FirebaseAnalytics;
+
+/**
+ * Library of Analytics events reported by Intra.  In addition to improving the reporting API,
+ * this class also attaches the Mobile Country Code (MCC) to each event, which indicates the client
+ * country, to compensate for country classification errors in Firebase.
+ */
+public class AnalyticsWrapper {
+
+  // Analytics events
+  private enum Events {
+    BOOTSTRAP,
+    BOOTSTRAP_FAILED,
+    BYTES,
+    EARLY_RESET,
+    STARTVPN,
+    TLS_PROBE,
+    TRY_ALL_ACCEPTED,
+    TRY_ALL_CANCELLED,
+    TRY_ALL_DIALOG,
+    TRY_ALL_FAILED,
+    TRY_ALL_REQUESTED,
+    UDP,
+  }
+
+  // Mobile Country Code, or 0 if it is unknown.
+  private final int mcc;
+  private final FirebaseAnalytics analytics;
+  private AnalyticsWrapper(int mcc, FirebaseAnalytics analytics) {
+    this.mcc = mcc;
+    this.analytics = analytics;
+  }
+
+  public static AnalyticsWrapper get(Context context) {
+    return new AnalyticsWrapper(MobileCountryCode.get(context),
+        FirebaseAnalytics.getInstance(context));
+  }
+
+  private void log(Events e, @NonNull BundleBuilder b) {
+    b.put(Params.MCC, mcc);
```

Comment:
We need a name here that conveys:

- It's an ISO country code, not an MCC
- It's the home operator and network operator, set when the user is connected to its home cell network

It would also be useful to better understand when those things are set. I see 3 states:
- user on cell network at home
- user on cell network while roaming
- user not on cell network (e.g. wifi)

What do you think about this:
- `CELL_HOME_COUNTRY`: the "sim operator country". Set if the phone has a SIM card or cdma configured
- `CELL_NETWORK_COUNTRY` (maybe `CELL_CONNECTED_COUNTRY`? `CELL_NET_COUNTRY`?): Empty if not available (not on a cell network), set to the country if available and == `CELL_HOME_COUNTRY` or set to `ZZ` if it differs from `CELL_HOME_COUNTRY`, for privacy

This way we'll be able to better understand when those things are set, and the meaning of the parameters will be clearer.

Reply:
I went with `COUNTRY` (the consensus country, or empty string if there is none) and `NETWORK` (the network type).  I think that's the minimal solution that gives us the necessary information and lets us exclude outlier events.